test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/DefaultEditorTemplatesTest.cs

```diff
@@ -916,7 +930,129 @@ public void Editor_AppliesRfc3339(string dataTypeName, string editFormatString,
                 Mock.Of<IUrlHelper>(),
                 viewEngine.Object,
                 provider);
-            helper.Html5DateRenderingMode = Html5DateRenderingMode.Rfc3339;
+            helper.Html5DateRenderingMode = Html5DateRenderingMode.CurrentCulture;
+            helper.ViewData.TemplateInfo.HtmlFieldPrefix = "FieldPrefix";
+
+            // Act
+            var result = helper.Editor(string.Empty);
+
+            // Assert
+            Assert.Equal(expectedInput, HtmlContentUtilities.HtmlContentToString(result));
+        }
+
+        // Html5DateRenderingMode.Rfc3339 is enabled by default.
+        [Theory]
+        [InlineData(null, null, "2000-01-02T03:04:05.060", "datetime-local")]
+        [InlineData("date", "{0:d}", "2000-01-02", "date")]
+        [InlineData("datetime", null, "2000-01-02T03:04:05.060", "datetime-local")]
+        [InlineData("datetime-local", null, "2000-01-02T03:04:05.060", "datetime-local")]
+        [InlineData("DateTimeOffset", "{0:o}", "2000-01-02T03:04:05.060Z", "text")]
```

Comment:
FYI this demonstrates user can opt into new `type` and format combination for `DateTime` properties.